Сделан список треков на экране поиска по трекам

amdoit-git opened the pull request.

amdoit-git committed:
Сделан список треков на экране поиска по трекам

IliaAnK commented on .idea/deploymentTargetDropDown.xml:
```diff
@@ -3,7 +3,20 @@
   <component name="deploymentTargetDropDown">
```
Рекомендация:
Этот файл стоит добавить в `gitignore` файл

IliaAnK commented on app/src/main/res/layout/track_in_list.xml:
```diff
@@ -0,0 +1,38 @@
+
+        <TextView
+            android:id="@+id/trackName"
+            style="@style/trackName"
```
Рекомендация:
Не нужно выносить свойства каждой View в отдельно стиль.
Стили нужны для унификации поведения и/или внешнего вида и переиспользования этих характеристик во множестве мест.

Любой стиль, который используется в этом файле, имеет специфические свойства, которые не дадут переиспользовать его, следовательно выделения таких характеристик в стиль абсолютно бесполезное, а в чём-то даже вредно.

Пример хорошего стиля:
Выделить определённый стиль текста. К примеру, это будет заголовок, у которого будет определённые шрифт, кегель, интерлиньяж, цвет и т. п.
И стиль этого заголовка можно применять к любой TextView.

Актуально для любого стиля, который используется в этом файле.

amdoit-git replied:
> Рекомендация: Не нужно выносить свойства каждой View в отдельно стиль. Стили нужны для унификации поведения и/или внешнего вида и переиспользования этих характеристик во множестве мест.
> 
> Любой стиль, который используется в этом файле, имеет специфические свойства, которые не дадут переиспользовать его, следовательно выделения таких характеристик в стиль абсолютно бесполезное, а в чём-то даже вредно.
> 
> Пример хорошего стиля: Выделить определённый стиль текста. К примеру, это будет заголовок, у которого будет определённые шрифт, кегель, интерлиньяж, цвет и т. п. И стиль этого заголовка можно применять к любой TextView.
> 
> Актуально для любого стиля, который используется в этом файле.

А можно ли в такой случае вообще все свойства View элементов в подобных лейаутах (которые вставляются в верстку программными средствами)  указывать в атрибутах? При условии, что они не повторяются в других лейаутах. Так на самом деле их проще редактировать - они все на одном экране находятся. При этот рендеринг таких элементов на устройстве будет быстрее или медленнее, чем при указании стилей?

IliaAnK commented on app/src/main/res/layout/track_in_list.xml:
```diff
@@ -0,0 +1,38 @@
+        <TextView
+            android:id="@+id/trackName"
+            style="@style/trackName"
+            android:text="@string/trackName" />
```
Рекомендация:
Для теста вёрстки стоит использовать неймспейс `tools`  вместо `android`.
Контент с неймспейс `tools` не попадёт в итоговую сборку, что защищает от возможного отображения тестовой информации.

И что ещё **важно**, не нужно выносить значения для теста вёрстки в ресурсы. 
По большому счёту, текст в ресурсы выносится для того, чтобы сформировать единый источник и можно было локализовывать приложение под разные языки. А если пользователь не увидит этот текст, то зачем его выносить в ресурсы?

IliaAnK commented on app/src/main/java/com/example/playlistmaker/TrackAdapter.kt:
```diff
@@ -0,0 +1,57 @@
+        holder.bind(tracks[position]);
+    }
+
+    class TrackHolder(view: View) : RecyclerView.ViewHolder(view) {
```
Рекомендация:
Класс `TrackHolder` лучше вынести в отдельный файл.
Не стоит вкладывать классы просто так.

amdoit-git committed:
Вложенный класс TrackHolder вынесен из внешнего класса.
Из ресурсов strings убраны строки использующиеся только для проверки верстки
Поля View элементов, которые не будут видны пользователю указаны через tools: вместо android:
